chezbetty/models/box.py

```diff
@@ -27,7 +28,25 @@ def from_id(cls, id):
 
     @classmethod
     def from_barcode(cls, barcode):
-        return DBSession.query(cls).filter(cls.barcode == barcode).one()
+        box_list = DBSession.query(cls).filter(cls.barcode.ilike('%{}%'.format(barcode))).all()
+        if(len(box_list) == 1):
+            return box_list[0]
+        else: #list length other than 1 is likely a bad scan, so search for exact match to throw exception if needed
+            return DBSession.query(cls).filter(cls.barcode == barcode).one()
+        #TODO: handle the two cases where len(box_list) != 1 (len is 0, len is > 1)
+
+    @classmethod #search for all delimited barcodes in database; return true if any matches are found
+    def update_exists_barcode(cls, barcode, id):
```

Comment:
Should this be `barcode_exists()`? It doesn't update does it?

Reply:
So this is what I called the function which is used when updating a box or item in the database, rather than creating a new item. The ID is required so that it doesn't falsely detect itself as a conflict, which is why there are separate functions. The naming might be odd, but I wasn't sure what else to call it...it is checking if the barcode exists in the admin_box_edit_submit and admin_edit_item_submit functions.